felix/bpf/proxy/syncer_test.go

```diff
@@ -1096,6 +1096,46 @@ var _ = Describe("BPF Syncer", func() {
 			Expect(eps.m).To(HaveLen(2))
 		}))
 
+		By("checking endpointslice terminating status should be included in endpointslice collection for processing", makestep(func() {
+			state.SvcMap[svcKey] = proxy.NewK8sServicePort(
+				net.IPv4(10, 0, 0, 1),
+				1234,
+				v1.ProtocolTCP,
+			)
+			state.EpsMap[svcKey] = []k8sp.Endpoint{
+				&k8sp.BaseEndpointInfo{Ready: true, Endpoint: "10.1.0.1:5555"},
+				&k8sp.BaseEndpointInfo{Terminating: true, Endpoint: "10.1.0.2:5555"},
+			}
+
+			err := s.Apply(state)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(eps.m).To(HaveLen(1))
+		}))
+
+		By("checking that conntrack scan does not remove the terminating endpoint connection", makestep(func() {
+			svc := state.SvcMap[svcKey]
+			eps := state.EpsMap[svcKey]
+			for _, ep := range eps {
+				if ep.IsReady() {
```

Comment:
We should either include the Terminating only or Ready and Terminating. Perhaps the latter is better to make sure that the change does not effect the Ready ones. But the **the key is to include the Teminating**, because that is what we want to make sure that does not get removed from the ct table!

So you could either find the terminating and create the entry for that one. Or create entry for all 3 and then make sure that you also find 3 after the clean up.

If you wrote the test before the rest of the changes, it would fail and now it should succeed after the change to the syncer code, which proves the change.

Reply:
Thanks @tomastigera - updated unit test but thought best actually to "start fresh" i.e. clean up all prior state and then follow suggestion i.e. create entry for all 3x and also followed the convention [here](https://github.com/projectcalico/calico/blob/master/felix/bpf/proxy/syncer_test.go#L174) i.e. manually unroll loop to add multiple conntrack entries - thanks